File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0;
        int count = 0;
        int res = 0;
        unordered_map<char, int> mp;
        for(int i = 0; i < s.size(); i++){
            mp[s[i]]++;
            count = max(mp[s[i]], count);
            while(i - l +1 - count > k){
                mp[s[l]]--;
                l++;
            }
            res = max( res, i -l +1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/maximum-difference-between-even-and-odd-frequency-i/submission-3.py
class Solution:
    def maxDifference(self, s: str) -> int:
        count = Counter(s)
        maxeven, maxodd = float('inf'), 0
        res = 0
        for v in count.values():
            if v % 2 == 0:
                maxeven = min(maxeven, v)
            else:
                maxodd = max(maxodd, v)
        return maxodd - maxeven